pkg/apis/crd/v1alpha2/types.go

```diff
@@ -205,6 +205,21 @@ type EgressStatus struct {
 	// EgressIP indicates the effective Egress IP for the selected workloads. It could be empty if the Egress IP in spec
 	// is not assigned to any Node. It's also useful when there are more than one Egress IP specified in spec.
 	EgressIP string `json:"egressIP"`
+
+	Conditions []EgressCondition `json:"conditions,omitempty"`
+}
+
+type EgressConditionType string
+
+const (
+	IPAllocated EgressConditionType = "IPAllocated"
+	IPAssigned  EgressConditionType = "IPAssigned"
+)
+
+type EgressCondition struct {
+	Type               EgressConditionType `json:"type"`
+	Status             v1.ConditionStatus  `json:"status"`
+	LastTransitionTime metav1.Time         `json:"lastTransitionTime"`
```

Comment:
I think we need Reason and Message to tell user why the condition is false so they know how to fix it.